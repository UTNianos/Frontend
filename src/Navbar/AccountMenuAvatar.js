import React from 'react';
import { Avatar } from 'antd';
import './AccountMenu/AccountMenu.css';

const AccountMenuAvatar = ({user}) => {

  if(user.avatar === undefined || user.avatar.trim() === '')
    return(
    <Avatar
      type="circle"
      size='medium'
      className="Avatar"
     >
      {user.username[0]}
    </Avatar>
    );

  const userAvatar = '/static/avatars/'+ user.avatar;

  return(
  <Avatar
    type="circle"
    size='medium'
    className="Avatar"
    src={userAvatar}
  />
  );

}

export default AccountMenuAvatar;
