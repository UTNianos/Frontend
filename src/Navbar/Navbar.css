.navbar-container {
  background-color: white;
  box-shadow: 0 2px 8px #f0f1f2;
}

.logo.container {
  margin-left: 200px;
  width: 140px;
}

.logo.container.mobile {
  margin-left: 0px;
  width: 100%;
}

.logo.container.mobile .header-search {
  margin-left: 40px;
}

.logo.container.mobile .menu-logo {
  padding-left: 35%;
}

.anticon.anticon-search,.ant-badge {
  color: gainsboro;
}

.anticon.anticon-search {
  margin-right: 20px;
}

.topnav.header-logo
{
  width: 165px;
  height: 64px;
  position: relative;
  line-height: 64px;
  transition: all .3s;
  overflow: hidden;
  color: gainsboro;
}

.topnav.header-logo img
{
  display: inline-block;
  vertical-align: middle;
  height: 32px;
}

.topnav.header-logo h1 {
  color: gainsboro;
  display: inline-block;
  vertical-align: top;
  font-size: 16px;
  margin: 0 0 0 12px;
  font-weight: 400;
}

.userDropdown
{
  margin-left: 30px;
  color: gainsboro;
}

.userName
{
  margin-left: 10px;
}

/* Sider styles */

.sider {
  min-height: 100vh;
  box-shadow: 2px 0 6px rgba(0, 21, 41, 0.35);
  position: relative;
  z-index: 10;
}

.sider.fixSiderbar {
    position: fixed;
    top: 0;
    left: 0;
}

.sider.light {
  box-shadow: 2px 0 8px 0 rgba(29, 35, 41, 0.05);
  background-color: white;
}
