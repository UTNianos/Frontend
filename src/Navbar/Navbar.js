import React, { PropTypes } from 'react';
import { Link } from 'react-router-dom';

const Navbar = () => (
  <ul>
    <li>
      <Link to="/">
		UTNianos
	   </Link>
    </li>
  </ul>
  );

export default Navbar;
