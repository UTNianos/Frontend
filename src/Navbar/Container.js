import { connect } from 'react-redux';
import Navbar from './Navbar';

const mapStateToProps = () => ({ userLoggedIn: false });
const mapDispatchToProps = () => ({});

const NavbarContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(Navbar);

export default NavbarContainer;
