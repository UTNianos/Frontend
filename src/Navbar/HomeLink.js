import React from 'react';
import { Link } from 'react-router-dom';
import logo from './logo_utnianos.png';
import './Navbar.css';

const HomeLink = () => (
  <div className="logo container">
    <Link to="/" className="topnav header-logo">
      <img src={logo} alt="UTNianos" />
    </Link>
  </div>
);

export default HomeLink;
