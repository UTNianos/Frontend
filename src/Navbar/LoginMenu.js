import React from 'react';
import { Button } from 'antd';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faSignInAlt as signIn,
  faUserPlus as userPlus
} from '@fortawesome/free-solid-svg-icons';
import { Link } from 'react-router-dom';
import './Navbar.css';

const LoginMenu = () => {
  return (
  <span className="LoginButtons">
		 <Link to="/login/login">
		   <Button type="primary" size="default" style={{marginRight:'4px'}}>
			   <FontAwesomeIcon icon={signIn} />
			   &nbsp;
			   Log in
		   </Button>
	   </Link>

	   <Link to="/login/register">
	     <Button type="primary" size="default">
		     <FontAwesomeIcon icon={userPlus} />
		     &nbsp;
		     Register
	     </Button>
	   </Link>
	</span>
  );
};

export default LoginMenu;
