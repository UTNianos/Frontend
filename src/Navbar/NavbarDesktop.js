import React from 'react';
import { Row, Col } from 'antd';
import logo from './logo_utnianos.png';
import './Navbar.css';

// Menus.
import LoginMenu from './LoginMenu';
import AccountMenu from './AccountMenu/Container';

const NavbarMenu = (props) => {

  const {
    logoutFn,
    loggedIn,
    notifications,
    user,
    clearFn
  } = props;

  if(loggedIn)
    return(
    <div style={{marginLeft:'176px'}}>
      {/*
        <Notifications
          notifications={notifications} clearFn={clearFn}
        />
       */}
      <AccountMenu user={user} logoutFn={logoutFn} />
    </div>
    )

   return <LoginMenu />;
}

const Navbar = (props) => {

  const {
    loggedIn,
    user,
    notifications,
    logoutFn,
    dismissNotifications,
  } = props;

  return (
  <div className="navbar-container">
    <Row gutter={16}>
      <Col className="gutter-row" span={6}>
        <div className="logo container">
          <a href="/" className="topnav header-logo">
            <img src={logo} alt="UTNianos" />
            <h1>UTNianos</h1>
          </a>
        </div>
      </Col>
      <Col className="gutter-row" span={6}>
      </Col>
      <Col className="gutter-row" span={6}></Col>
      <Col className="gutter-row" span={6}>
        <NavbarMenu
          notifications={notifications}
          loggedIn={loggedIn}
          user={user}
          logoutFn={logoutFn}
          clearFn={dismissNotifications}
        />
      </Col>
    </Row>
  </div>
  );
}

/*
Navbar.propTypes = {
  : PropTypes.
};*/

export default Navbar;
