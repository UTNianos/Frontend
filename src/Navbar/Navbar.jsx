import React from 'react'
import NavbarDesktop from './NavbarDesktop';
import NavbarMobile from './NavbarMobile';
import MediaQuery from 'react-responsive';
import { Affix } from 'antd';

const Navbar = () => {

  return (
  <Affix>
   <MediaQuery minDeviceWidth={1224} >
      <NavbarDesktop />
    </MediaQuery>
	<MediaQuery maxDeviceWidth={1224}>
      <NavbarMobile />
    </MediaQuery>
  </Affix>
  );
}

export default Navbar
