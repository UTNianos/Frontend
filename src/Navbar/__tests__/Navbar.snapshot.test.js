import React from 'react';
import renderer from 'react-test-renderer';
import Navbar from '../Navbar';

jest.mock('react-dom');
jest.mock('antd');

describe('<Navbar />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Navbar userLoggedIn={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  })
});
