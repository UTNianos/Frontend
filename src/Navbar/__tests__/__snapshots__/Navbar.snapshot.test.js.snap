// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginMenu /> Render correcto. 1`] = `
<span>
  <a
    href="/login"
    onClick={[Function]}
  />
  <a
    href="/register"
    onClick={[Function]}
  />
</span>
`;

exports[`<Navbar /> Render correcto. 1`] = `null`;
